app/dashboard/views.py

```diff
@@ -1133,9 +1139,111 @@ def profile(request, handle):
+    order_by = request.GET.get('order_by', '-modified_on')
+    if context['preferred_payout_address']:
+        owned_kudos = Token.objects.filter(
+            owner_address=to_checksum_address(context['preferred_payout_address']),
+            contract__network=settings.KUDOS_NETWORK
+        ).order_by(order_by)
+        sent_kudos = Token.objects.filter(
+            kudos_transfer__from_address=to_checksum_address(context['preferred_payout_address']),
+            contract__network=settings.KUDOS_NETWORK
+        ).order_by(order_by)
+    else:
+        owned_kudos = None
+        sent_kudos = None
+
+    if owned_kudos:
+        owned_kudos_comments_public = []
+        for kudos_token in owned_kudos:
+            try:
+                owned_kudos_comments_public.append(kudos_token.kudos_transfer.comments_public)
+            except Exception as e:
+                logger.error(e)
+        logging.info(f'kudos comments_public: {owned_kudos_comments_public}')
+    else:
+        owned_kudos_comments_public = None
+    context['kudos_comments_public'] = owned_kudos_comments_public
+    context['kudos'] = owned_kudos
+    context['sent_kudos'] = sent_kudos
+
+    if request.method == 'POST' and request.is_ajax():
+        # Send kudos data when new preferred address
+        address = request.POST.get('address')
+        context['kudos'] = Token.objects.filter(owner_address=to_checksum_address(address)).order_by(order_by)
+        context['sent_kudos'] = Token.objects.filter(kudos_transfer__from_address=to_checksum_address(address)).order_by(order_by)
+        profile.preferred_payout_address = address
+
+        kudos_html = loader.render_to_string(
+            'shared/profile_kudos.html',
+            context
+        )
+        # package output data and return it as a JSON object
+        output_data = {
+            'kudos_html': kudos_html,
+        }
+
+        try:
+            profile.save()
+        except Exception as e:
+            logger.error(e)
+            msg = {
+                'status': 500,
+                'msg': 'Internal server error',
+            }
+        else:
+            wallet_addresses = []
+            wallet_addresses.append(profile.preferred_payout_address)
+            msg = {
+                'status': 200,
+                'msg': 'Success!',
+                'wallets': wallet_addresses,
+                'kudos_html': kudos_html,
+            }
+
+        return JsonResponse(msg)
 
     return TemplateResponse(request, 'profiles/profile.html', context, status=status)
 
+def lazy_load_kudos(request):
+    page = request.POST.get('page')
+    address = request.POST.get('address')
+    context= {}
+    # with this attr from the button clicked we know if we request kudos or sent kudos
+    datarequest = request.POST.get('request')
+    order_by = request.GET.get('order_by', '-modified_on')
+    # use Django’s pagination
+    # https://docs.djangoproject.com/en/dev/topics/pagination/
```

Comment:
Can we omit these comments?  Also, should we use `paginator.get_page(page)` vs the manual handling of emptypage/integer/etc ?

Reply:
We can evaluate, I don't really know if with get_page we can do the same, I have seen get_page looks simpler. Right now each request is controlled by an ajax request from front end and then is returned the html cards json that a way (not nearly the best I know) to avoid redo all the front end logic two times (when page loads and when load more is clicked)  In that way we are reusing the django template (quicker)  instead  of fill html with js. I think in a V2 we should go to a full ajax > json way and then change to get_page

Reply:
So I was taking a look go get_page and I think doesn't make change to wait to change, code gets more simpler, making the changes here!